queue/group_remote_mongo.go

```diff
@@ -196,22 +200,30 @@ func (g *remoteMongoQueueGroup) Queues(ctx context.Context) []string {
 }
 
 func (g *remoteMongoQueueGroup) startProcessingRemoteQueue(ctx context.Context, coll string, opts ...amboy.QueueOptions) (amboy.Queue, error) {
+	id := g.idFromCollection(coll)
+	// The driver already adds the jobs suffix implicitly to the collection
+	// name.
 	coll = trimJobsSuffix(coll)
 	mdbOpts, err := getMongoDBQueueOptions(opts...)
 	if err != nil {
 		return nil, errors.Wrap(err, "getting queue options")
 	}
 
-	queueOpts := mergeMongoDBQueueOptions(append([]MongoDBQueueOptions{g.opts.Queue}, mdbOpts...)...)
+	precedenceOrderedOpts := []MongoDBQueueOptions{g.opts.DefaultQueue}
+	if perQueueOpts, ok := g.opts.PerQueue[id]; ok {
+		precedenceOrderedOpts = append(precedenceOrderedOpts, perQueueOpts)
+	}
+	precedenceOrderedOpts = append(precedenceOrderedOpts, mdbOpts...)
```

Comment:
So just to be clear: the order of precedence is (1) mdb options, (2) per que opts, and (3) default?

Reply:
Yes.